lib/auth/api_with_roles.go

```diff
@@ -77,12 +78,35 @@ func (api *APIWithRoles) Serve() {
 	for role := range api.listeners {
 		wg.Add(1)
 		go func(listener net.Listener, handler http.Handler) {
+			defer wg.Done()
 			if err := http.Serve(listener, handler); (err != nil) && (err != io.EOF) {
-				log.Errorf(err.Error())
+				if !api.askedToStop {
+					log.Errorf(err.Error())
+				}
 			}
 		}(api.listeners[role], api.servers[role])
 	}
 	wg.Wait()
+	log.Infof("[AUTH] API exited")
+}
+
+func (api *APIWithRoles) Close() {
+	api.askedToStop = true
```

Comment:
it's used to suppress an unwanted log message after `http.Serve()` fails due to a force-closed socket.